chatkit-core/src/test/kotlin/com/pusher/chatkit/users/state/CurrentUserReceivedTest.kt

```diff
@@ -0,0 +1,54 @@
+package com.pusher.chatkit.users.state
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import com.pusher.chatkit.state.CurrentUserReceived
+import com.pusher.chatkit.state.State
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+
+class CurrentUserReceivedTest : Spek({
+
+    val simpleUser = UserInternalType(
+        id = "danielle",
+        createdAt = 1583422281000L,
+        updatedAt = 1583422302000L,
+        name = "daniellevass",
+        avatarUrl = null,
+        customData = null
+    )
+
+    describe("given initial state") {
+        val initialState = State.initial()
+
+        describe("when a current user is received") {
+            val currentUserReceived = CurrentUserReceived(simpleUser)
+            val newState = currentUserReceivedReducer(initialState, currentUserReceived)
+
+            it("then the state contains the current user") {
+                assertThat(newState.currentUser).isEqualTo(simpleUser)
+            }
+        }
+    }
+
+    describe("given a current user") {
```

Comment:
is this test even useful? it's unlikely to happen IRL.

Reply:
It will happen each time there is a new `initial_state` from the user subscription (because the subscription was interrupted and recreated)

Reply:
Hm. but does that mean the currentUser would change at all? We'd expect for the id to be the same? But perhaps they've updated the avatar or custom data - at which point we should just overwrite the old object?

Reply:
The current user ID should not change, but the other fields might, correct.

Reply:
`createdAt` should not change as well I reckon?